script/libraries/LibInitializeGuard.sol

```diff
@@ -0,0 +1,321 @@
+   * - Check if `_initialized` slot is found.
+   * - Check if the last initialized version is equal to the number of `initialize` functions.
+   */
+  function _validateProxiesVersion(Cache storage $cache) private {
+    address[] memory proxies = $cache._proxies.values();
+    uint256 length = proxies.length;
+
+    for (uint256 i; i < length; ++i) {
+      address proxy = proxies[i];
+      InitializedSlot memory slot = $cache._initSlot[proxy];
+
+      require(
+        slot.found,
+        string.concat("LibInitializeGuard: Proxy ", vm.getLabel(proxies[i]), " does not have `_initialized` slot!")
+      );
+
+      uint256 lastInitVer = $cache._lastInitVer[proxy];
+
+      require(
+        lastInitVer != 0, string.concat("LibInitializeGuard: Proxy ", vm.getLabel(proxy), " does not initialize!")
+      );
+
+      if (
+        (lastInitVer == MAX_VER_V4 && slot.nBit == N_BIT_INIT_V4)
+          || (lastInitVer == MAX_VER_V5 && slot.nBit == N_BIT_INIT_V5)
+      ) {
+        string memory ret = vm.prompt(
+          string.concat(
+            "[WARNING] ".yellow(),
+            vm.getLabel(proxy),
+            " disabled initialized version.".yellow(),
+            " Is it intentional?\n".yellow(),
+            "Press ",
+            "yes ".blue(),
+            "to continue..."
+          )
+        );
+        require(keccak256(bytes(vm.toLowercase(ret))) == keccak256("yes"), "LibInitializeGuard: User aborted!");
+
+        continue;
+      }
+
+      uint256 initFnCount = _getInitializeFnCount($cache, proxy);
+      require(
+        lastInitVer == initFnCount,
+        string.concat(
+          "LibInitializeGuard: Invalid initialized version!",
+          " Expected: ",
+          vm.toString(initFnCount),
+          " Got: ",
+          vm.toString(lastInitVer)
+        )
+      );
+    }
+  }
+
+  /**
+   * @dev Validate the intermediate changes of the `_initialized` slot of the given `access` storage.
+   *
+   * @param acc The storage access of data.
+   * @param slot The initialized slot of the proxy.
+   * @return shouldSkip Whether to skip the validation.
+   */
+  function _validateInitChanges(Vm.StorageAccess memory acc, InitializedSlot memory slot)
+    private
+    view
+    returns (bool shouldSkip)
+  {
+    uint256 mask = (1 << slot.nBit) - 1;
+
+    uint256 prvVer = (uint256(acc.previousValue) >> slot.bitOffset) & mask;
+    uint256 newVer = (uint256(acc.newValue) >> slot.bitOffset) & mask;
+
+    // Skip if `_initialized` bytes location in `slot` does not change
+    // Assume other data in given slot is not related to initialized version
+    if (prvVer == newVer) return true;
+    // Skip if the proxy disable initialized version
+    if ((newVer == MAX_VER_V4 && slot.nBit == N_BIT_INIT_V4) || (newVer == MAX_VER_V5 && slot.nBit == N_BIT_INIT_V5)) {
+      console.log("[INIT] %s: Disabled initialized version", vm.getLabel(acc.account));
+      return true;
+    }
```

Comment:
If I unintentionally disable initialized at proxy contract, this validation could be bypassed and this guard coundnt warn me? Correct me if I'm wrong.

Reply:
This guard will kindly warn u:
https://github.com/axieinfinity/foundry-deployment-kit/blob/implement-feature/rework/add-initialize-guard/script/libraries/LibInitializeGuard.sol#L157